src/structures/GuildChannel.js

```diff
@@ -414,18 +414,18 @@ class GuildChannel extends Channel {
    * @param {string} [options.reason] Reason for cloning this channel
    * @returns {Promise<GuildChannel>}
    */
-  clone({ name = this.name, withPermissions = true, withTopic = true, nsfw, parent, bitrate, userLimit, reason } = {}) {
-    const options = {
-      overwrites: withPermissions ? this.permissionOverwrites : [],
-      nsfw: typeof nsfw === 'undefined' ? this.nsfw : nsfw,
-      parent: parent || this.parent,
-      bitrate: bitrate || this.bitrate,
-      userLimit: userLimit || this.userLimit,
-      reason,
+  clone(options = {}) {
+    const createOptions = {
+      overwrites: options.overwrites ? this.permissionOverwrites : [],
```

Comment:
This should be `options.withPermissions` instead.
And it should default to true as it did before.

Reply:
Whoops. You're right. And what code pattern should I use to default undefined to true? I can revert the changes to the destructured object back (although I changed it to this because of the possibility of needing to add more keys in the future).

Reply:
Options coming to my mind are
- [mergeDefault](https://github.com/hydrabolt/discord.js/blob/master/src/util/Util.js#L122) - feels slightly redundant
- `(thing || typeof thing === 'undefined') ? val1 : val2` - would exceed max line length
- Destructuring object - would exceed max line length with more keys in the future

I probably tend to the last one, but it does not _has_ to be that one though.
Maybe someone else has a better idea 👀

Reply:
To keep consistency in key defaulting, `mergeDefault` is actually good idea, it ensures keys are defaulted correctly and keeps the code clean as it doesn't bloat the method's code with defaults. After this, it'd be as easy as passing the options with the defaults applied to the createOptions object for later usage in [GuildChannelStore#create()](https://discord.js.org/#/docs/main/master/class/GuildChannelStore?scrollTo=create).

However, at this point it'd make no sense to create a new object (createOptions) and pass it to the aforementioned method as it filters them: [here](https://github.com/hydrabolt/discord.js/blob/e576387fea2a973eefe298a6f3610eb0d8b54999/src/stores/GuildChannelStore.js#L52).

After that update, `GuildChannel#clone()` would be probably just 3-4 lines long.